corehq/apps/users/models.py

```diff
@@ -1673,6 +1674,8 @@ def unretire(self):
 
         deleted_case_ids = self._get_deleted_case_ids()
         CaseAccessors(self.domain).soft_undelete_cases(deleted_case_ids)
+
+        undelete_system_forms(self.domain, set(deleted_form_ids), set(deleted_case_ids))
```

Comment:
Could this have unintended behavior?  Could there be forms which were intentionally deleted that will now be reinstated?

Reply:
I don't think so.

should this be called with `.delay()`?

Reply:
doh, yes it should, updated.